Task1.5

Sarah0711 opened the pull request.

Jemyz committed:
updates git ignore

Jemyz committed:
initialize task 1.5

Sarah0711 committed:
Data

Jemyz committed:
updates task 1.5

Jemyz committed:
.Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Sarah0711 committed:
Generate data (hypersphere)

mmghannam commented on project1/task1.5.py:
```diff
@@ -0,0 +1,4 @@
+import project1.strategy_kmeans as sk
+import numpy as np
+if __name__ == '__main__':
+    sk.strategy_kmeans( np.array([[1, 2], [1, 4], [1, 0],[10, 2], [10, 4], [10, 0]]))
```
Shouldn't we apply it on the data that is generated, or am I understanding something wrong?

mmghannam commented on project1/Data.py:
```diff
@@ -1,22 +1,25 @@
-
-plt.show()
\ No newline at end of file
+def Data(k,dim,n):
```
Shouldn't this have a name relating to a sphere in some way?

mmghannam commented on project1/data.out:
```diff
@@ -0,0 +1,270 @@
+3.134497669826219468e-01,1.834758356002318402e-01
```
I assume these are the data points that are generated to be a hypersphere. Do we need this file? as I don't see it used anywhere. and also, what's the difference between data.out and data1.out?

Sarah0711 committed:
Update Data.py

Jemyz committed:
implements kmeans method

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Sarah0711 committed:
Update Data.py

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Sarah0711 committed:
Update Data.py

Sarah0711 committed:
Update Data.py

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Sarah0711 committed:
Update Data.py

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Sarah0711 committed:
Update Data.py

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Jemyz committed:
implements kmeans task

Sarah0711 committed:
Data generation methos modified with use of convex hull. Addedfew generated examples

Sarah0711 committed:
Removing previous data which was not correct for use

Sarah0711 committed:
Changes address when radius-the sum of other coordinates were negative

Sarah0711 committed:
New data generation for the second strategy

Sarah0711 committed:
New Data for n=1000

Jemyz committed:
fixs bugs in kmeans

Jemyz committed:
Merge branch 'task1.5' of github.com:mmghannam/Pattern-Recognition-II-Projects into task1.5

Jemyz committed:
implements convex hull strategy

Sarah0711 committed:
1.Changes with respect to facematrix data
2.Dimensionality reduction with PCA then application of convex hull strategy
3.Changed distancematrix to cdist to consider euclidean distance and for error manhattan distance
4.Visualisation of extracted farthest data
5.Evaluate function to evaluate strategies with own data and given data with different variations

Sarah0711 committed:
1.Results
2.Restructure the data files

Jemyz committed:
reformats the code

Jemyz committed:
implements all of the figures

Jemyz committed:
removes some comments

Jemyz committed:
fixes plots bug

Jemyz committed:
adds new data

Jemyz committed:
fixes some bugs
